clang/tools/scan-build-py/lib/libscanbuild/intercept.py

```diff
@@ -270,6 +270,17 @@ def abspath(cwd, name):
         fullname = name if os.path.isabs(name) else os.path.join(cwd, name)
         return os.path.normpath(fullname)
 
+# SYCLomatic_CUSTOMIZATION begin
+    def get_object_files(flags):
+        return [flag for flag in flags if flag.endswith('.o')]
+
+    def filter_linker_entry(flags):
+        object_files = get_object_files(flags)
+        output_file_idx = flags.index('-o')
+        output_flags = flags[output_file_idx:output_file_idx+2]
```

Comment:
Pls and comment for magic number "2", and also make sure the code style consistent with current script code style, for example there should be one space before and after operator "+".